src/include/mpichconfconst.h

```diff
@@ -60,11 +60,9 @@
 #define MPICH_LOGGING__EXTERNAL 4
 
 /* Possible values for process state */
-#define MPICH_MPI_STATE__PRE_INIT 0
-#define MPICH_MPI_STATE__POST_INIT 1
-#define MPICH_MPI_STATE__POST_FINALIZED 2
-/* Internal init states. */
-#define MPICH_MPI_STATE__IN_INIT 3      /* can call MPIR_Err_return_comm(...) on error */
+#define MPICH_MPI_STATE__UNINITIALIZED     0
+#define MPICH_MPI_STATE__LOCAL_INITIALIZED 1
+#define MPICH_MPI_STATE__INITIALIZED       2
```

Comment:
Per MPI 4.0 standard, `MPI_Session_init` is a local function. This PR calls `MPII_Init_thread`, which is a global collective, thus is strictly incorrect. Although for any reasonable practical MPI application, all process will call `MPI_Sessioin_init` or one of the init anyway, so this PR probably will satisfy all practical usages. However, since it is strictly incorrect implementation, next PR will split the current "init" into strictly "local" init -- where we don't do any address exchange or communication, and a "global/collective" init -- where we do address exchange and creating communicators. `MPICH_MPI_STATE__LOCAL_INITIALIZED` is preparing for that.

Reply:
On further thoughts, we actually don't need the `MPICH_MPI_STATE__LOCAL_INITIALIZED` to track local init state. We could just use `MPICH_MPI_STATE__INITIALIZED`. The world init state can and should be checked by whether `MPIR_Process.comm_world` is set -- I should enforce that in the later PRs.

However, we do need a state to signal (currently only the error utilities) that we finished mpr-layer init. I have renamed the state to `MPICH_MPI_STATE__MPIR_INITIALIZED`, I think that reflects the nature much better.